R/analyze_binary.R

```diff
@@ -96,32 +87,46 @@ deepstate_fuzz_fun_analyze<- function(test_function,seed=-1, time.limit.seconds,
   runner_harness_name <- paste0(runner_line_split[2], "_", runner_line_split[4])
   
   run.executable <- if(seed == -1 && time.limit.seconds != -1){
-    paste0("cd ",test_function," && valgrind --xml=yes --xml-file=",log_file,
-           " --tool=memcheck --leak-check=yes --track-origins=yes ",
-           "./",basename(test_function),"_DeepState_TestHarness --timeout=",time.limit.seconds,
-           " --fuzz --input_which_test ",runner_harness_name," > ",valgrind.log.text," 2>&1")
+    paste0("cd ",test_function," && valgrind --xml=yes --xml-file=",
+           valgrind_log_xml, " --tool=memcheck --leak-check=yes ",
+           "--track-origins=yes ", "./", basename(test_function),
+           "_DeepState_TestHarness --timeout=", time.limit.seconds, 
+           " --fuzz --input_which_test ", runner_harness_name," > ",
+           valgrind_log_txt," 2>&1")
   }else{
-    paste0("cd ",test_function," && valgrind --xml=yes --xml-file=",log_file,
-           " --tool=memcheck --leak-check=yes --track-origins=yes ",
-           "./",basename(test_function),"_DeepState_TestHarness --seed=",seed,
-           " --timeout=",time.limit.seconds," --fuzz --input_which_test ",runner_harness_name,
-           " > ",valgrind.log.text," 2>&1")
+    paste0("cd ",test_function," && valgrind --xml=yes --xml-file=",
+           valgrind_log_xml, " --tool=memcheck --leak-check=yes ",
+           "--track-origins=yes ", "./",basename(test_function),
+           "_DeepState_TestHarness --seed=",seed, " --timeout=",
+           time.limit.seconds," --fuzz --input_which_test ",
+           runner_harness_name, " > ",valgrind_log_txt," 2>&1")
   }
   if (verbose){
     message(sprintf("running the executable .. \n%s\n", run.executable))
   }
-  system(run.executable, ignore.stdout=!verbose)
+  exit_code <- system(run.executable, ignore.stdout=!verbose)
+  valgrind_log_content <- readLines(valgrind_log_txt)
+  fuzzing_crash <- all(grepl("Starting fuzzing", valgrind_log_content)==FALSE)
+  if (exit_code != 0 && fuzzing_crash){
+    error_msg <- paste("The function", fun_name, "has not been analyzed due to",
+                       "some errors while running the test harness. You can",
+                       "find more details inside", valgrind_log_txt)
+    stop(error_msg)
+  }
+
+
   for(inputs.i in seq_along(inputs.path)){
     file.copy(inputs.path[[inputs.i]],output_folder)
     if(grepl(".qs",inputs.path[[inputs.i]],fixed = TRUE)){
-      inputs_list[[gsub(".qs","",basename(inputs.path[[inputs.i]]))]] <- qread(inputs.path[[inputs.i]])
+      inputs_file <- gsub(".qs","",basename(inputs.path[[inputs.i]]))
+      inputs_list[[inputs_file]] <- qread(inputs.path[[inputs.i]])
     }else{
-      inputs_list[[basename(inputs.path[[inputs.i]])]] <-scan(inputs.path[[inputs.i]],quiet = TRUE)
+      inputs_file <- basename(inputs.path[[inputs.i]])
+      inputs_list[[inputs_file]] <-scan(inputs.path[[inputs.i]],quiet = TRUE)
```

Comment:
space after assignment operator please: ` <- ` instead of ` <-`.
also maybe use an intermediate variable `input.i.base <- basename(inputs.path[[inputs.i]])`

Reply:
also do we even need the non-qs (scan) branch anymore? I thought all of the data was written via qs files?

Reply:
Yes we still need that branch because of the `arma::mat` datatype. Qs doesn't support this datatype, thus the only way to save it is using a binary file. 

I added an intermediate variable and refactored the code with commit e8e20d5